goss-testing/scripts/data-validator.py

```diff
@@ -0,0 +1,289 @@
+#!/usr/bin/env python3
+#
+# MIT License
+#
+# (C) Copyright 2022 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+import re
+import subprocess
+import sys
+import ipaddress
+import socket
+import json
+import os
+
+
+def print_err(*a):
+  print(*a, file = sys.stderr)
+
+
+def get_data():
```

Comment:
If this function hits an error (for example, if it cannot reach BSS, or if data.json has a formatting issue), then it prints an error message and the get_data function returns 1. But the overall Python script returns exit code 0 in this case.  And the Goss test is only looking at the exit code, not stderr, so it would also pass.

Given that an error in this function means that the rest of the test really cannot run, it seems like it would be better to call **sys.exit(1)** rather than **return 1**, in case of problems.

Reply:
Agreed. There is no reason to go past that point if that function fails.